### PR TITLE
add a CRL public_bytes method

### LINKED ISSUE #2540: Support Serializing CRLs
We need a `public_bytes` method...

### PR DESCRIPTION
**Pull request by reaperhulk**
Fixes #2540

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -16,6 +16,9 @@ Changelog
 * :class:`~cryptography.x509.NameConstraints` are now supported in the
   :class:`~cryptography.x509.CertificateBuilder` and
   :class:`~cryptography.x509.CertificateSigningRequestBuilder`.
+* Support serialization of certificate revocation lists using the
+  ``public_bytes`` method of
```
Comment (alex) [inline CHANGELOG.rst]: Is there a reason not to make this a link?

File: tests/test_x509.py [hunk 2]
```diff
@@ -222,6 +222,48 @@ def test_tbs_certlist_bytes(self, backend):
...
+        crl = _load_cert(
+            os.path.join("x509", "custom", "crl_empty.pem"),
+            x509.load_pem_x509_crl,
+            backend
+        )
+
+        # Encode it to PEM and load it back.
+        crl = x509.load_pem_x509_crl(crl.public_bytes(
+            encoding=serialization.Encoding.PEM,
+        ), backend)
```
Comment (alex) [inline tests/test_x509.py]: Should we check the exact bytes here

File: tests/test_x509.py [hunk 3]
```diff
@@ -222,6 +222,48 @@ def test_tbs_certlist_bytes(self, backend):
...
+        crl = _load_cert(
+            os.path.join("x509", "custom", "crl_all_reasons.pem"),
+            x509.load_pem_x509_crl,
+            backend
+        )
+
+        # Encode it to DER and load it back.
+        crl = x509.load_der_x509_crl(crl.public_bytes(
+            encoding=serialization.Encoding.DER,
+        ), backend)
```
Comment (alex) [inline tests/test_x509.py]: Same.

File: tests/test_x509.py (hunk 2 above)
Comment (reaperhulk) [inline tests/test_x509.py]: Added a test that does that for PEM/DER (same as we use for cert public_bytes)

**Pull request merged**